base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -73,6 +71,11 @@
      */
     public SoftButtonObject(@NonNull String name, @NonNull List<SoftButtonState> states, @NonNull String initialStateName, OnEventListener onEventListener) {
 
+        // If the list of states is empty, throw an error with DebugTool and return
+        if (states.isEmpty()) {
+            DebugTool.logError(TAG,"The state list is empty");
+            return;
+        }
```

Comment:
Alignment note:

On iOS I did this

```objc
BOOL hasStateWithInitialName = NO;
for (SDLSoftButtonState *state in states) {
    if ([state.name isEqualToString:initialStateName]) {
        hasStateWithInitialName = YES;
    }
}
NSAssert(![self sdl_hasTwoStatesOfSameName:states], @"A SoftButtonObject must have states with different names.");
NSAssert(hasStateWithInitialName, @"A SoftButtonObject must have a state with initialStateName.");
if ([self sdl_hasTwoStatesOfSameName:states]) { return nil; }
if (!hasStateWithInitialName) { return nil; }
```

The advantage this has is only throwing an exception in DEBUG builds (I don't know that is available on Java Suite). Second, this tests not only that there _are_ states, but also that the initial state name is available. The for loop won't run if there are no states, so the boolean will just stay false. That check at least should be a part of this PR as well.

Last, I also made a change to `SoftButtonState` on iOS to check if both text and image are `null`, which is an invalid state configuration. I think that should be a part of this PR as well.

Reply:
Understood, I'll work on implementing this